src/tools/opensc-explorer.c

```diff
@@ -1469,6 +1478,92 @@ static int do_get(int argc, char **argv)
 	return -err;
 }
 
+static int do_get_record(int argc, char **argv)
+{
+	u8 buf[SC_MAX_EXT_APDU_RESP_SIZE];
+	int r, err = 1;
+	size_t rec, count = 0;
+	sc_path_t path;
+	sc_file_t *file = NULL;
+	char *filename;
+	FILE *outf = NULL;
+
+	if (argc < 2 || argc > 3)
+		return usage(do_get);
+	if (arg_to_path(argv[0], &path, 0) != 0)
+		return usage(do_get);
+	rec = strtoul(argv[1], NULL, 10);
+
+	filename = (argc == 3) ? argv[2] : path_to_filename(&path, '_', rec);
+	outf = (strcmp(filename, "-") == 0)
+		? stdout
+		: fopen(filename, "wb");
+	if (outf == NULL) {
+		perror(filename);
+		goto err;
+	}
+
+	r = sc_lock(card);
+	if (r == SC_SUCCESS)
+		r = sc_select_file(card, &path, &file);
+	sc_unlock(card);
```

Comment:
sc_unlock behind sc_read_record to bind the read file to the selected one?

Reply:
@frankmorgner  I am sorry, I do not understand.  This piece of code was copied verbatim from `do_get()`, and the pattern highlighted in these four lines is used all across `opensc-explorer.c`.

Reply:
Technically there is no problem at all. However, if you want to avoid being deselected between the SELECT and the GET RECORD, then you may want to lock before SELECT and unlock when the GET RECORD is done. This may also improve the pattern for the other operations, but I'll leave it to you whether or not to tie the both commands together.